querybuilder.go

```diff
@@ -98,30 +98,32 @@ func (qb QueryBuilder) Filter(qo queryoptions.Options) (bson.M, error) {
 					f := primitive.M{field: bv}
 					filter = combine(filter, f)
 				}
-			case "date":
-				f := detectDateComparisonOperator(field, values)
-				filter = combine(filter, f)
-			case "decimal":
-				f := detectNumericComparisonOperator(field, values, bsonType)
-				filter = combine(filter, f)
-			case "double":
-				f := detectNumericComparisonOperator(field, values, bsonType)
-				filter = combine(filter, f)
-			case "int":
-				f := detectNumericComparisonOperator(field, values, bsonType)
-				filter = combine(filter, f)
-			case "long":
-				f := detectNumericComparisonOperator(field, values, bsonType)
-				filter = combine(filter, f)
-			case "object":
-				f := detectStringComparisonOperator(field, values, bsonType)
-				filter = combine(filter, f)
-			case "string":
-				f := detectStringComparisonOperator(field, values, bsonType)
+			case "date", "timestamp":
+				var f bson.M
+				if len(values) == 1 {
+					f = detectDateComparisonOperator(field, values)
+				} else {
```

Comment:
There is a scenario here I'm testing with locally that's giving an unanticipated result (there is a unit test for it also in the querybuilder_test.go). If I provide something like `filter[val]=1,2,3,4` (the intended goal would be to do an $in search as of now), it turns the query into the $and search you've adjusted to. This is a breaking change from the previous version of the library, but I have an idea...

What if the values were each examined to understand if there is a desired operator (i.e. ne, lte, gte, lt or gt) in any of the list values, and if so, the $and is used instead of the $in. This way, the old functionality would exist without change, but the new desired functionality you've proposed would continue to work as expected. I'm testing a tweak on my local fork of your changes to see if I can get this working, and I'll reply this afternoon with some further thoughts!

Reply:
That sounds like a much better solution indeed, I basically just did this as a temporary fix since we needed it solved on short notice. Another alternative might be to utilize a different operator for the and operations (=**+**>=100,<=10)

I'd be happy to look over what you've come up with later on (tomorrow for me) as well! Happy you like the concept

Reply:
Just a quick update: I'm close... but I need another day I think. Considering having some sort of strategy setting (defaulting to $and) that can be used to allow users to specify $or when needed. The time I'm spending now is refining the query syntax and building units.